src/console/panel.ts

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import {
+  ISession
+} from 'jupyter-js-services';
+
+import {
+  Message
+} from 'phosphor-messaging';
+
+import {
+  Widget
+} from 'phosphor-widget';
+
+import {
+  TabPanel
+} from 'phosphor-tabs';
+
+import {
+  SplitPanel
+} from 'phosphor-splitpanel';
+
+import {
+  showDialog
+} from '../dialog';
+
+import {
+  RenderMime
+} from '../rendermime';
+
+import {
+  ConsolePager
+} from './pager';
+
+import {
+  ConsoleTooltip
+} from './tooltip';
+
+import {
+  ConsoleWidget
+} from './widget';
+
+
+/**
+ * The class name added to console panels.
+ */
+const CONSOLE_PANEL = 'jp-ConsolePanel';
+
+
+/**
+ * A panel which contains a toolbar and a console.
```

Comment:
Yeah, that was the original plan and never happened. I only just updated that comment.